src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4235,7 +4220,10 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             else
             {
                 TypeSymbolWithAnnotations rightType = VisitOptionalImplicitConversion(right, leftType, UseLegacyWarnings(left), AssignmentKind.Assignment);
-                TrackNullableStateForAssignment(right, leftType, MakeSlot(left), rightType, MakeSlot(right));
+                int rightSlot = MakeSlot(right);
+                int slotWatermark = GetSlotWatermark();
+                int leftSlot = MakeSlot(left);
```

Comment:
@jcouv, I've added a comment based on our discussion and logged [https://github.com/dotnet/roslyn/issues/33429](https://github.com/dotnet/roslyn/issues/33429).